pygmt/src/nearneighbor.py

```diff
@@ -131,6 +133,18 @@ def nearneighbor(data=None, x=None, y=None, z=None, **kwargs):
         - :class:`xarray.DataArray`: if ``outgrid`` is not set
         - None if ``outgrid`` is set (grid output will be stored in file set by
           ``outgrid``)
+    Example
+    -------
+    >>> import pygmt
+    >>> # Load a sample dataset of bathymetric x, y, and z values
+    >>> data = pygmt.datasets.load_sample_data(name="bathymetry")
+    >>> # Create a new grid with 5m spacing in the designated region
```

Comment:
Could you please update the docstring to explain what `search_radius="10m"` does?